### PR TITLE
feat: interval measurements and counters

### LINKED ISSUE #57: Pulse counter does not reset timer prescaler before it starts counting
The [pulse counter ](https://github.com/fossasia/pslab-firmware/blob/feb2acd0ee48178ca62b74def8607cd01a45d79f/PSLab_Original/proto2_main.c#L902) uses timer2 to count pulses on a digital input. Some other functions also use timer2, notably the logic analyzer in four channel mode, which supports setting a prescaler to slow down the timer.

If a prescaler is set by the logic analyzer, the pulse counter will count slower than expected. It should reset the timer2 prescaler before it starts counting.

### PR DESCRIPTION
**Pull request by CloudyPadmal**
Also fixes #57

### CODE REVIEW

File: pslab-core.X/instruments/sensors.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+     * 
+     * @param pin1 First pin connected to a Digital pin
+     * @param pin2 Second pin connected to a Digital pin
+     * @param pin1_edge Triggering transition on pin 1
+     * @param pin2_edge Triggering transition on pin 2
+     * 
+     * @return None
+     */
+    void SENSORS_ConfigureInterval(uint8_t pin1, uint8_t pin2,
+            uint8_t pin1_edge, uint8_t pin2_edge);
```
Comment (bessman) [inline pslab-core.X/instruments/sensors.h:18]: Type should be `IC_PARAMS_CAPTURE_MODE`, right?

File: pslab-core.X/instruments/sensors.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    
+    /**
+     * @brief Count the rise and fall instances of a signal connected to a 
+     * digital pin
+     *
```
Comment (bessman) [inline pslab-core.X/instruments/sensors.h:23]: On some other command functions I have also documented the args and return vals the function takes/returns over serial. I think that's helpful in understanding what the function does. What do you think?

File: pslab-core.X/instruments/sensors.h (hunk 1 above)
Comment (CloudyPadmal) [reply pslab-core.X/instruments/sensors.h:18]: Yes! This needs to be updated.

File: pslab-core.X/instruments/sensors.h (hunk 2 above)
Comment (CloudyPadmal) [reply pslab-core.X/instruments/sensors.h:23]: I agree. Let me update the documentation bit more..

**Review by bessman — APPROVED** [review]

**Pull request merged**